pkg/psmdb/mongo/mongo.go

```diff
@@ -445,6 +445,39 @@ func (m *ConfigMembers) RemoveOld(compareWith ConfigMembers) (changes bool) {
 	return changes
 }
 
+// ExternalNodesChanged checks if votes or priority fields changed for external nodes
+func (m *ConfigMembers) ExternalNodesChanged(compareWith ConfigMembers) (changes bool) {
+	cm := make(map[string]struct {
+		votes    int
+		priority int
+	}, len(compareWith))
+
+	for _, member := range compareWith {
+		_, ok := member.Tags["external"]
+		if !ok {
+			continue
+		}
+		cm[member.Host] = struct {
+			votes    int
+			priority int
+		}{votes: member.Votes, priority: member.Priority}
+	}
+
+	for i := 0; i < len(*m); i++ {
+		member := []ConfigMember(*m)[i]
+		if ext, ok := cm[member.Host]; ok {
+			changes = (ext.votes != member.Votes || ext.priority != member.Priority)
+			[]ConfigMember(*m)[i].Votes = ext.votes
+			[]ConfigMember(*m)[i].Priority = ext.priority
+			if changes {
```

Comment:
we can't just make a return there.
As i understand we need to update list `m` with all `compareWith` values

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/729/commits/0ffbd4f700fa3d2ba415029012c99d39493d5202